server/controllers/resourceParserController.js

```diff
@@ -6,16 +7,43 @@ const resourceParserController = {
   parsePut: async (req, res, next) => {
     
     try{
-
-        
-      res.locals.newResource = newResource;
-      return next();
+      
+      const msgObject = await req.body[0];
+      const keyArr = ["id", "user", "date", "text", "embeds", "attachments", "score", "resources", "subject", "category"];
+      const hasKeys = await keyArr.every(key => msgObject.hasOwnProperty(key));
```

Comment:
This is great, covers all of the property checks in a single line, but is still really easy to understand, nice! You can probably shorten even more by dropping the await, I don't think every() returns a promise.